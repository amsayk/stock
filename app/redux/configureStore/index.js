// @flow

import { createStore, applyMiddleware, compose } from 'redux';

import { APP_NAME, DEBUG, INIT } from 'vars';

import pick from 'lodash.pick';

import array from 'redux/middleware/array';
import bufferActions from 'redux/middleware/buffer-actions';

import { createHashHistory } from 'history';
import { routerMiddleware, routerActions } from 'react-router-redux';
import locationHelperBuilder from 'redux-auth-wrapper/history4/locationHelper';

import { client as apolloClient } from 'apollo';

import { createSnackbarController } from 'components/Snackbar';

import thunk from 'redux-thunk';

import rootReducer from 'redux/reducers';

const history = createHashHistory();

const configureStore = (initialState?: any) => {
  // ======================================================
  // Middleware Configuration
  // ======================================================
  const middlewares = [
    bufferActions(),
    array,
    thunk.withExtraArgument({
      client: apolloClient,
      history,
      get snackbar() {
        return (
          store.snackbar || (store.snackbar = createSnackbarController(store))
        );
      },
    }),
    routerMiddleware(history),
  ];

  // Redux DevTools Configuration
  const actionCreators = {
    ...routerActions,
  };

  const composeEnhancers =
    (__DEV__ || DEBUG) && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
      ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
          name: APP_NAME,
          routerActions,
        })
      : compose;

  // ======================================================
  // Store Enhancers
  // ======================================================
  const enhancers = [applyMiddleware(...middlewares)];

  const enhancer = composeEnhancers(...enhancers);

  const store = createStore(rootReducer, enhancer);

  // Snackbar
  store.snackbar = createSnackbarController(store);

  // Initialize
  store.dispatch({ type: INIT });

  if (__DEV__) {
    if (module.hot) {
      module.hot.accept('../reducers', () => {
        store.replaceReducer(require('../reducers'));
      });
    }
  }

  return store;
};

export default {
  configureStore,
  history,
  locationHelper: locationHelperBuilder({}),
};
