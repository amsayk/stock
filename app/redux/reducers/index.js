import user from './user/reducer';
import app from './app/reducer';
import snackbar from './snackbar/reducer';
import notification from './notification/reducer';
import scrolling from './scrolling/reducer';
import toastr from './toastr/reducer';
import intl from './intl/reducer';

import { USER_LOGGED_OUT } from './user/constants';

import { combineReducers } from 'redux-immutable';

import { INIT } from 'vars';

import { routerReducer as router } from 'react-router-redux';

import { reducer as formReducer, actionTypes } from 'redux-form/immutable';

const reducers = {
  intl,
  app,
  notification,
  scrolling,
  user,
  snackbar,
  toastr,
  form: formReducer.plugin({
    login: (state, action) => {
      switch (action.type) {
        case USER_LOGGED_OUT:
          return state
            ? state.mergeDeep({
                values: {
                  password: undefined,
                },
              })
            : state;
        case actionTypes.SUBMIT_START: // Clear errors
          return state
            ? state.merge({
                error: null,
              })
            : state;
        default:
          return state;
      }
    },
    changePassword: (state, action) => {
      switch (action.type) {
        case actionTypes.SET_SUBMIT_SUCCEEDED: // Clear passwords on success
          return state
            ? state.merge({
                values: {},
              })
            : state;
        default:
          return state;
      }
    },
    changeEmail: (state, action) => {
      switch (action.type) {
        case actionTypes.SET_SUBMIT_SUCCEEDED: // Clear email on success
          return state
            ? state.merge({
                values: {},
              })
            : state;
        default:
          return state;
      }
    },
  }),
};

export default combineReducers({
  router,
  ...reducers,
});
