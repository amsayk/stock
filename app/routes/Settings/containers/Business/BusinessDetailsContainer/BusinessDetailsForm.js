import React from 'react';
import T from 'prop-types';

import { compose } from 'redux';

import { withApollo } from 'react-apollo';

import checkBusiness from 'utils/checkBusiness';

import {
  reduxForm,
  Field,
  propTypes as reduxFormPropTypes,
  SubmissionError,
} from 'redux-form/immutable';

import ImmutablePropTypes from 'react-immutable-proptypes';

import isEmpty from 'isEmpty';

import { intlShape } from 'react-intl';

import messages from 'routes/Settings/messages';

import style from 'routes/Settings/styles';

import validations from './validations';

import validationMessages from 'validation-messages';

import BusinessNameField from 'routes/Settings/components/BusinessNameField';
import BusinessDescriptionField from 'routes/Settings/components/BusinessDescriptionField';
import OptionalTextInputField from 'routes/Settings/components/OptionalTextInputField';
import CountryField from 'routes/Settings/components/CountryField';

import MUTATION from './updateUserBusiness.mutation.graphql';

import update from 'react-addons-update';

export class BusinessDetailsForm extends React.Component {
  constructor(props, context) {
    super(props, context);

    this.onKeyDown = this._onKeyDown.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }
  _onKeyDown(e) {
    if (e.key === 'Enter' && e.shiftKey === false) {
      const submit = this.props.handleSubmit(this.onSubmit);
      submit();
    }
  }

  async onSubmit(data) {
    try {
      await validations.asyncValidate(data);
    } catch (e) {
      throw new SubmissionError(e);
    }

    const { intl } = this.props;

    const {
      data: { updateUserBusiness: { errors } },
    } = await this.props.client.mutate({
      mutation: MUTATION,
      variables: {
        payload: {
          displayName: data.get('displayName'),
          description: data.get('description'),
          url: data.get('url'),
          country: data.get('country'),
          addressLine1: data.get('addressLine1'),
          addressLine2: data.get('addressLine2'),
          city: data.get('city'),
          stateProvince: data.get('stateProvince'),
          postalCode: data.get('postalCode'),
          phone: data.get('phone'),
          taxId: data.get('taxId'),
        },
      },
      updateQueries: {
        getUser: (prev, { mutationResult }) => {
          const newOrUpdatedBusiness =
            mutationResult.data.updateUserBusiness.business;
          return isEmpty(errors)
            ? update(prev, {
                currentUser: {
                  business: {
                    $set: newOrUpdatedBusiness,
                  },
                },
              })
            : prev;
        },
      },
    });

    if (!isEmpty(errors)) {
      throw new SubmissionError(errors);
    }

    const { snackbar } = this.context;
    if (snackbar) {
      snackbar.show({
        message: intl.formatMessage(messages.businessUpdateSuccessNotification),
      });
    }

    // currentUser.business has changed, refresh notification.
    await checkBusiness();
  }

  render() {
    const { intl, handleSubmit, pristine, submitting, invalid } = this.props;
    return (
      <div className={style.content}>
        <h1 className={style.formHeading}>
          {intl.formatMessage(messages.linkBusinessDetails)}
        </h1>
        <div className={style.form}>
          <Field
            name="displayName"
            component={BusinessNameField}
            label={intl.formatMessage(messages.labelBusinessName)}
            onKeyDown={this.onKeyDown}
          />
          <Field
            name="description"
            component={BusinessDescriptionField}
            label={intl.formatMessage(messages.labelBusinessDescription)}
          />
          <Field
            name="url"
            component={OptionalTextInputField}
            label={intl.formatMessage(messages.labelURL)}
            onKeyDown={this.onKeyDown}
          >
            <div className={style.formControlFeedback} when={'webSite'}>
              {intl.formatMessage(validationMessages.urlInvalid)}
            </div>
          </Field>
          <Field
            name="country"
            component={CountryField}
            label={intl.formatMessage(messages.labelCountry)}
            onKeyDown={this.onKeyDown}
          />
          <Field
            name="addressLine1"
            component={OptionalTextInputField}
            label={intl.formatMessage(messages.labelAddressLine1)}
            onKeyDown={this.onKeyDown}
          />
          <Field
            name="addressLine2"
            component={OptionalTextInputField}
            label={intl.formatMessage(messages.labelAddressLine2)}
            onKeyDown={this.onKeyDown}
          />
          <Field
            name="city"
            component={OptionalTextInputField}
            label={intl.formatMessage(messages.labelCity)}
            onKeyDown={this.onKeyDown}
          />
          <Field
            name="stateProvince"
            component={OptionalTextInputField}
            label={intl.formatMessage(messages.labelStateProvince)}
            onKeyDown={this.onKeyDown}
            className={style.width15Percent}
          />
          <Field
            name="postalCode"
            component={OptionalTextInputField}
            label={intl.formatMessage(messages.labelPostalCode)}
            onKeyDown={this.onKeyDown}
            className={style.width15Percent}
          >
            <div className={style.formControlFeedback} when={'zipCode'}>
              {intl.formatMessage(validationMessages.zipInvalid)}
            </div>
          </Field>
          <Field
            name="phone"
            component={OptionalTextInputField}
            label={intl.formatMessage(messages.labelPhone)}
            onKeyDown={this.onKeyDown}
          >
            <div className={style.formControlFeedback} when={'phoneNumber'}>
              {intl.formatMessage(validationMessages.phoneInvalid)}
            </div>
          </Field>
          <Field
            name="taxId"
            component={OptionalTextInputField}
            label={intl.formatMessage(messages.labelTaxId)}
            onKeyDown={this.onKeyDown}
          />
          <div className={style.formGroup}>
            <div className={style.offset2} />
            <button
              onClick={handleSubmit(this.onSubmit)}
              disabled={pristine || submitting || invalid}
              className={style.saveButton}
            >
              {intl.formatMessage(messages.save)}
            </button>
          </div>
        </div>
      </div>
    );
  }
}

BusinessDetailsForm.contextTypes = {
  snackbar: T.shape({
    show: T.func.isRequired,
  }),
};

BusinessDetailsForm.defaultProps = {};

BusinessDetailsForm.propTypes = {
  ...reduxFormPropTypes,
  intl: intlShape.isRequired,
  user: ImmutablePropTypes.contains({
    id: T.any.isRequired,
  }).isRequired,
  initialValues: ImmutablePropTypes.contains({
    id: T.string,
    displayName: T.string,
    description: T.string,
    url: T.string,
    country: T.string,
    addressLine1: T.string,
    addressLine2: T.string,
    city: T.string,
    stateProvince: T.string,
    postalCode: T.string,
    phone: T.string,
    taxId: T.string,
  }).isRequired,
};

const Form = reduxForm({
  form: 'business.details',
  enableReinitialize: true,
  keepDirtyOnReinitialize: false,
});

export default compose(withApollo, Form)(BusinessDetailsForm);
