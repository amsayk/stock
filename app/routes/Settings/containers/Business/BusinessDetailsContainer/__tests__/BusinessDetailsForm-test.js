import React from 'react';
import { compose } from 'redux';
import configureMockStore from 'redux-mock-store';
import { reduxForm } from 'redux-form/immutable';
import { IntlProvider, injectIntl } from 'react-intl';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

import { fromJS } from 'immutable';

import { BusinessDetailsForm } from '../BusinessDetailsForm';

const client = {};

const Decorated = compose(injectIntl, reduxForm({ form: 'testForm' }))(
  BusinessDetailsForm,
);

const middlewares = [];
const mockStore = configureMockStore(middlewares);

describe('business details form', () => {
  const store = mockStore(
    fromJS({
      form: {
        testForm: {},
      },
    }),
  );

  it('should render the snapshot', () => {
    const tree = renderer
      .create(
        <IntlProvider
          defaultLocale={'en'}
          locale={'en'}
          messages={{}}
          formats={{}}
        >
          <Provider store={store}>
            <Decorated
              user={{
                id: 'id',
              }}
              client={client}
              initialValues={{}}
            />
          </Provider>
        </IntlProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
