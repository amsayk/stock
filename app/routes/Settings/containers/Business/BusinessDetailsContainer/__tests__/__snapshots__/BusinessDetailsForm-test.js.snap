exports[`business details form should render the snapshot 1`] = `
<div
  className="content">
  <h1
    className="formHeading">
    Business details
  </h1>
  <div
    className="form">
    <div
      className="formGroup">
      <label
        className="label"
        htmlFor="displayName">
        Business name
         
        <span
          className="iconRequired">
          ⋆
        </span>
      </label>
      <div
        className="inputWrapper">
        <input
          autoComplete="off"
          autoFocus={true}
          className="control"
          name="displayName"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          type="text"
          value="" />
        <div />
      </div>
    </div>
    <div
      className="formGroup">
      <label
        className="label"
        htmlFor="description">
        Description
      </label>
      <div
        className="inputWrapper">
        <textarea
          className="control"
          name="description"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          rows={5}
          style={
            Object {
              "resize": "none",
            }
          }
          value="" />
      </div>
    </div>
    <div
      className="formGroup">
      <label
        className="label"
        htmlFor="url">
        Web site
      </label>
      <div
        className="inputWrapper">
        <input
          className="control"
          name="url"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          value="" />
        <div />
      </div>
    </div>
    <div
      className="formGroup">
      <label
        className="label"
        htmlFor="country">
        Country
         
        <span
          className="iconRequired">
          ⋆
        </span>
      </label>
      <div
        className="inputWrapper">
        <select
          className="control"
          disabled={true}
          name="country"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          value="">
          <option
            value="US">
            United States
          </option>
        </select>
        <div />
      </div>
    </div>
    <div
      className="formGroup">
      <label
        className="label"
        htmlFor="addressLine1">
        Address line 1
      </label>
      <div
        className="inputWrapper">
        <input
          className="control"
          name="addressLine1"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          value="" />
        <div />
      </div>
    </div>
    <div
      className="formGroup">
      <label
        className="label"
        htmlFor="addressLine2">
        Address line 2
      </label>
      <div
        className="inputWrapper">
        <input
          className="control"
          name="addressLine2"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          value="" />
        <div />
      </div>
    </div>
    <div
      className="formGroup">
      <label
        className="label"
        htmlFor="city">
        City
      </label>
      <div
        className="inputWrapper">
        <input
          className="control"
          name="city"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          value="" />
        <div />
      </div>
    </div>
    <div
      className="formGroup">
      <label
        className="label"
        htmlFor="stateProvince">
        State / Province
      </label>
      <div
        className="inputWrapper">
        <input
          className="control width15Percent"
          name="stateProvince"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          value="" />
        <div />
      </div>
    </div>
    <div
      className="formGroup">
      <label
        className="label"
        htmlFor="postalCode">
        ZIP / Postalcode
      </label>
      <div
        className="inputWrapper">
        <input
          className="control width15Percent"
          name="postalCode"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          value="" />
        <div />
      </div>
    </div>
    <div
      className="formGroup">
      <label
        className="label"
        htmlFor="phone">
        Phone
      </label>
      <div
        className="inputWrapper">
        <input
          className="control"
          name="phone"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          value="" />
        <div />
      </div>
    </div>
    <div
      className="formGroup">
      <label
        className="label"
        htmlFor="taxId">
        Tax ID
      </label>
      <div
        className="inputWrapper">
        <input
          className="control"
          name="taxId"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          value="" />
        <div />
      </div>
    </div>
    <button
      className="saveButton"
      disabled={true}
      onClick={[Function]}>
      Save
    </button>
  </div>
</div>
`;
